elastic/integrations
# [authentik] Initial release of the authentik

## Proposed commit message

Create New integration package authentik.

- Added event, user and group data stream.
- Added data collection logic for all the data stream.
- Added the ingest pipeline for all the data stream.
- Mapped fields according to the ECS schema and added Fields metadata in the appropriate yml files.
- Added dashboards and visualizations.
- Added test for pipeline for all the data stream.
- Added system test cases for all the data stream.



## Checklist


## How to test this PR locally

- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/authentik directory.
- Run the following command to run tests.
> elastic-package test
```
--- Test results for package: authentik - START ---
╭───────────┬─────────────┬───────────┬───────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE   │ DATA STREAM │ TEST TYPE │ TEST NAME                                 │ RESULT │ TIME ELAPSED │
├───────────┼─────────────┼───────────┼───────────────────────────────────────────┼────────┼──────────────┤
│ authentik │ event       │ pipeline  │ (ingest pipeline warnings test-event.log) │ PASS   │ 302.352163ms │
│ authentik │ event       │ pipeline  │ test-event.log                            │ PASS   │ 1.110027695s │
│ authentik │ group       │ pipeline  │ (ingest pipeline warnings test-group.log) │ PASS   │ 292.426896ms │
│ authentik │ group       │ pipeline  │ test-group.log                            │ PASS   │ 150.652255ms │
│ authentik │ user        │ pipeline  │ (ingest pipeline warnings test-user.log)  │ PASS   │ 284.673304ms │
│ authentik │ user        │ pipeline  │ test-user.log                             │ PASS   │ 353.384632ms │
╰───────────┴─────────────┴───────────┴───────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: authentik - END   ---
Done
--- Test results for package: authentik - START ---
╭───────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE   │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├───────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ authentik │ event       │ system    │ default   │ PASS   │ 40.304661465s │
│ authentik │ group       │ system    │ default   │ PASS   │  36.56547032s │
│ authentik │ user        │ system    │ default   │ PASS   │ 36.804686156s │
╰───────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: authentik - END   ---
Done
--- Test results for package: authentik - START ---
╭───────────┬─────────────┬───────────┬──────────────────────────┬────────┬──────────────╮
│ PACKAGE   │ DATA STREAM │ TEST TYPE │ TEST NAME                │ RESULT │ TIME ELAPSED │
├───────────┼─────────────┼───────────┼──────────────────────────┼────────┼──────────────┤
│ authentik │ event       │ static    │ Verify sample_event.json │ PASS   │ 134.849408ms │
│ authentik │ group       │ static    │ Verify sample_event.json │ PASS   │  126.02694ms │
│ authentik │ user        │ static    │ Verify sample_event.json │ PASS   │ 116.549649ms │
╰───────────┴─────────────┴───────────┴──────────────────────────┴────────┴──────────────╯
--- Test results for package: authentik - END   ---
Done
--- Test results for package: authentik - START ---
╭───────────┬─────────────┬───────────┬────────────────────────────────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE   │ DATA STREAM │ TEST TYPE │ TEST NAME                                                          │ RESULT │ TIME ELAPSED │
├───────────┼─────────────┼───────────┼────────────────────────────────────────────────────────────────────┼────────┼──────────────┤
│ authentik │             │ asset     │ dashboard authentik-490ec869-2ac1-4c30-9653-7916748d4f84 is loaded │ PASS   │      1.616µs │
│ authentik │             │ asset     │ dashboard authentik-e3c4a5f9-e47c-4cde-b755-859122e7ef80 is loaded │ PASS   │        251ns │
│ authentik │             │ asset     │ search authentik-16fb2a4f-720c-416f-9713-dfc87ce0cb79 is loaded    │ PASS   │        227ns │
│ authentik │             │ asset     │ search authentik-cfd98a0c-37de-40a7-a785-72104f99c515 is loaded    │ PASS   │        285ns │
│ authentik │             │ asset     │ search authentik-cfe038cb-a2ed-494f-a7ee-40723b96c029 is loaded    │ PASS   │        267ns │
│ authentik │ event       │ asset     │ index_template logs-authentik.event is loaded                      │ PASS   │        192ns │
│ authentik │ event       │ asset     │ ingest_pipeline logs-authentik.event-0.1.0 is loaded               │ PASS   │        226ns │
│ authentik │ group       │ asset     │ index_template logs-authentik.group is loaded                      │ PASS   │        213ns │
│ authentik │ group       │ asset     │ ingest_pipeline logs-authentik.group-0.1.0 is loaded               │ PASS   │        177ns │
│ authentik │ user        │ asset     │ index_template logs-authentik.user is loaded                       │ PASS   │        231ns │
│ authentik │ user        │ asset     │ ingest_pipeline logs-authentik.user-0.1.0 is loaded                │ PASS   │        161ns │
╰───────────┴─────────────┴───────────┴────────────────────────────────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: authentik - END   ---
Done
```




## Related issues


- Closes #10509 

## Screenshots

![Integration Page](https://github.com/user-attachments/assets/da096bb3-0463-4d40-83d4-d8e8ac310bf0)
![Overview Page](https://github.com/user-attachments/assets/8dcd1331-dd38-4cac-9cc9-1eca6e761863)
